Use issue forms to streamline the issue submission process

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request converts the existing product submission template to instead be a [GitHub issues form](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms). It also adds a complementary form for product update requests.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Using issue forms will both streamline the experience for users (especially those who are not as familiar with GitHub) as well as helping ensure the information needed is included.

### Please note ###

Additional configuration to auto-apply labels and auto-assign can be added.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. When looking at the two forms in the GitHub UI ([product-submission-form.yml](https://github.com/cisagov/log4j-affected-db/blob/improvement/use_issue_forms/.github/ISSUE_TEMPLATE/product-submission-form.yml) and [product-update-form.yml](https://github.com/cisagov/log4j-affected-db/blob/improvement/use_issue_forms/.github/ISSUE_TEMPLATE/product-update-form.yml)) they show no errors and the preview renders correctly.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

mcdonnnj committed:
Use an issue form for product submissions

Switch to using a GitHub Issues form for product submission issues. This
will provide a smoother interface for users to submit products to the
database and ensure that certain values are included with a submission.

mcdonnnj committed:
Add an issue form for product updates

Provide an issue form for product updates to complement the one for
product submissions. This will encourage people to follow the specific
workflows for submissions and updates.

mcdonnnj committed:
Rename template files to reflect that they are now forms

mcdonnnj committed:
Add missing input to issue forms

Added an input to provide information about the date of a product's
last update.

marcindulak commented on .github/ISSUE_TEMPLATE/product-submission-form.yml:
```diff
@@ -0,0 +1,78 @@
+---
+name: Submit a Product
+description: Submit a product to the database
+title: "[Product Submission]: <vendor> - <product>"
```
For a quick overview, it would be helpful if the title contained also `<status>`

mcdonnnj replied:
Since the submitter must fill out those values manually we are encouraging only the most important items in the title.

marcindulak commented on .github/ISSUE_TEMPLATE/product-update-form.yml:
The submission and update forms are similar, with the update form less strict, with some validations skipped, and the `product-vendor` and `product-name` inputs skipped as well. Would it technically help to combine the two forms into one `[Product information]` form, with an additional, required `submission-type` input settable to "Product submission" or "Product update" added?

mcdonnnj replied:
When adding a new product there is important information that is required, but there is no need to force a submitter to fill out information that isn't changing when submitting a request to update a product.

dav3r replied:
> When adding a new product there is important information that is required, but there is no need to force a submitter to fill out information that isn't changing when submitting a request to update a product.

I agree with this wholeheartedly.  The less information that users submit, the fewer opportunities there are for errors.

jsf9k reviewed (approved):
Very nice!

dav3r commented on .github/ISSUE_TEMPLATE/product-submission-form.yml:
I think it might improve usability if the explanatory blurb for each "status" option was closer to the "Status" dropdown menu (e.g. directly above it).

mcdonnnj replied:
Please see 41e536e.

dav3r replied:
Thanks!

dav3r commented on .github/ISSUE_TEMPLATE/product-submission-form.yml:
Do we want to give a couple of examples here in the description, e.g. "<2.0" or ">3.0, <3.6"?  

The main thing I'm hoping to avoid is people including the name of the product in this field (e.g. "Log4j 2.15.0" vs. just "2.15.0").

mcdonnnj replied:
Please see 230b4c9.

dav3r commented on .github/ISSUE_TEMPLATE/product-submission-form.yml:
Why are the `product-updated` options quoted, but the `product-status` options are not?

mcdonnnj replied:
GitHub interprets `Yes`/`No` as booleans and throws an error informing you that they must be quoted. I can quote the others for consistency if preferred.

mcdonnnj replied:
GitHub interprets `Yes`/`No` as booleans and throws an error informing you that they must be quoted. I can quote the others for consistency if preferred.

dav3r replied:
That makes sense.  I vote to quote the others for consistency, but I'm fine either way.

jsf9k replied:
I think the others are fine unquoted, but I am not prepared to fight to the death over the issue.

mcdonnnj replied:
Please see 0804f1e.

dav3r replied:
Excellent solution!

dav3r commented on .github/ISSUE_TEMPLATE/product-submission-form.yml:
I think this warrants a comma, but feel free to disagree.
```suggestion
      description: If an update is available, where can it be found?
```

jsf9k replied:
I think this wording is slightly better, but I'm fine either way:
```suggestion
      description: Where an update can be found, if one is available.
```

mcdonnnj replied:
Please see c5c6c68.

dav3r commented on .github/ISSUE_TEMPLATE/product-update-form.yml:
Same as previous comment:

```suggestion
      description: If an update is available, where can it be found?
```

jsf9k replied:
I think this wording is slightly better, but I'm fine either way:
```suggestion
      description: Where an update can be found, if one is available.
```

mcdonnnj replied:
Please see c5c6c68.

dav3r reviewed (approved):
Approved, but I left some comments, suggestions, and questions.  Very nice use of GitHub forms here! 🚀

mcdonnnj committed:
Adjust markdown element in product submission form

Move the markdown element that explains available statuses down so it
appears close to where a user is selecting the status. Given how form
elements are rendered it has been adjusted to appear after the dropdown
itself.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Adjust a description in the product submission form

Adjust the product name description to be more similar to other
descriptions.

mcdonnnj committed:
Update dropdown in the product submission form

Update the product update dropdown's label and options. Mainly focused
on removing usage of Yes/No because these are boolean values in YAML
and thus needed special handling compared to other strings.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add placeholders in the product submission form

Add placeholders for some of the required inputs in the form. This will
be most helpful for the product version, but for completeness they have
also been added for the product vendor and name.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Adjust markdown element in product update form

Move the markdown element that explains available statuses down so it
appears close to where a user is selecting the status. Given how form
elements are rendered it has been adjusted to appear after the dropdown
itself. This mirrors changes made in the product submission form.

mcdonnnj committed:
Add inputs to product update form

Add product vendor and product name inputs to the update form. This
will ensure that even if a submitter does not update the title we
capture this information.

mcdonnnj committed:
Add missing punctuation in description in product update form

mcdonnnj committed:
Add a placeholder to the product issue forms

Add a placeholder value for the last updated input in both the product
submission and product update issue forms. This will encourage the
appropriate timestamp format.

mcdonnnj committed:
Update dropdown in the product update form

Update the product update dropdown's label and options. Mainly focused
on removing usage of Yes/No because these are boolean values in YAML
and thus needed special handling compared to other strings. This mirrors
changes done to the product submission form.

mcdonnnj committed:
Update description for product update link in forms

Update the description for the product update link input in
both the product submission and product update forms.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

jsf9k commented on .github/ISSUE_TEMPLATE/product-submission-form.yml:
```diff
@@ -0,0 +1,82 @@
+    id: product-versions
+    attributes:
+      label: Product version(s)
+      description: What version(s) of the product are affected?
```
Very minor suggestion.  Feel free to disregard if your opinion differs.
```suggestion
      description: What version(s) of the product is (are) affected?
```

jsf9k reviewed (approved):
I made one very minor suggestion, but feel free to disregard it if you don't agree.

Great work here!

dav3r reviewed (approved):
This looks excellent to me! 🦅

mcdonnnj committed:
Update input description in the product submission form

Update the description for the product version input so that it fully
accounts for multiple versions.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Merge branch 'develop' into improvement/use_issue_forms

justmurphy reviewed (approved):
Looks great!  Thank you.
